writer.go

```diff
@@ -232,7 +232,7 @@ func (w *Writer) CompressFile(oldfile *os.File, cmpname string) error {
 		}
 		return err
 	}
-	return os.Remove(cmpname + ".tmp") // remove *.log.tmp file
```

Comment:
we should remove the tmp file at the end.

Reply:
the tmp file is the oldfile, I could modify the description as: `CompressFile compress log file write into .gz, closes oldFile and remove source file`, but I figured it would be best to handle oldFile in the parent function and also delete the file there. See line 288